### PR TITLE
Add ui tests for copying / moving to share folders

### LINKED ISSUE #2029: Add ui test coverage for file copying to share folder
As described, automate the below scenarios: 

- User can share a file to a new shared folder and delete original file
- User can share a file to a new shared folder and keep original file
- User can share a file to existing shared folder and delete original file
- User can share a file to existing shared folder and keep original file

### PR DESCRIPTION
**Pull request by asnaith**
closes #2029

adds coverage for the 4 scenarios when copying / moving files to new / existing shares and preserving / deleting the original

### CODE REVIEW

**Review by FSM1 — APPROVED** [review]
great stuff 🥇

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 1]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
       editSharedFolderModal.body().should("be.visible")
       editSharedFolderModal.userPermissionDropDown().should("contain.text", "edit")
     })
+
+    it("can copy a file to a new shared folder and preserve original", () => {
+      cy.web3Login({ deleteShareBucket: true, clearCSFBucket: true })
+
+      // upload a file and store it's name as cypress alias
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]:
```suggestion
      // upload a file and store its name as cypress alias
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 2]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      homePage.uploadFile("../fixtures/uploadedFiles/logo.png")
+      homePage.fileItemName().invoke("text").as("fileName")
+
+      // share file to newly created share folder
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]:
```suggestion
      // share file to newly created shared folder
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 3]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      })
+    })
+
+    it("can copy a file to a pre-existing shared folder and preserve original", () => {
+      cy.web3Login({ deleteShareBucket: true, clearCSFBucket: true })
+
+      navigationMenu.sharedNavButton().click()
+      sharedPage.createSharedFolder()
+
+      // upload a file and store it's name as cypress alias
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]:
```suggestion
      // upload a file and store its name as cypress alias
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 4]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      navigationMenu.homeNavButton().click()
+      homePage.uploadFile("../fixtures/uploadedFiles/logo.png")
+      homePage.fileItemName().invoke("text").as("fileName")
+
+      // share file to newly created share folder
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]:
```suggestion
      // share file to newly created shared folder
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 5]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      })
+    })
+
+    it("can move a file to a pre-existing shared folder and delete original", () => {
+      cy.web3Login({ deleteShareBucket: true, clearCSFBucket: true })
+
+      navigationMenu.sharedNavButton().click()
+      sharedPage.createSharedFolder()
+
+      // upload a file and store it's name as cypress alias
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]:
```suggestion
      // upload a file and store its name as cypress alias
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 6]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      navigationMenu.homeNavButton().click()
+      homePage.uploadFile("../fixtures/uploadedFiles/logo.png")
+      homePage.fileItemName().invoke("text").as("fileName")
+
+      // share file to newly created share folder
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]:
```suggestion
      // share file to newly created shared folder
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 7]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      // go to the new share and ensure the file was moved there
+      navigationMenu.sharedNavButton().click()
+      sharedPage.sharedFolderItemName().contains(sharedFolderName)
+        .should("be.visible")
+        .dblclick()
+      sharedPage.fileItemRow().should("have.length", 1)
+
+      // ensure file name of moved file is correct
+      cy.get("@fileName").then(($fileName) => {
+        sharedPage.fileItemName().contains(`${$fileName}`).should("be.visible")
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]: A little more elegant IMHO. Feel free to ignore if you feel it's too complex
```suggestion
      cy.get<string>("@fileName").then((fileName) => {
        sharedPage.fileItemName().contains(fileName).should("be.visible")
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 8]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      // go to the new share and ensure file was copied there
+      navigationMenu.sharedNavButton().click()
+      sharedPage.sharedFolderItemName().contains(sharedFolderName)
+        .should("be.visible")
+        .dblclick()
+      sharedPage.fileItemRow().should("have.length", 1)
+
+      // ensure file name of copied file is correct
+      cy.get("@fileName").then(($fileName) => {
+        sharedPage.fileItemName().contains(`${$fileName}`).should("be.visible")
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]: A little more elegant IMHO. Feel free to ignore if you feel it's too complex
```suggestion
      cy.get<string>("@fileName").then((fileName) => {
        sharedPage.fileItemName().contains(fileName).should("be.visible")
```

File: packages/files-ui/cypress/tests/file-sharing-spec.ts [hunk 9]
```diff
@@ -166,5 +170,152 @@ describe("File Sharing", () => {
...
+      // go to the new share and ensure file was moved there
+      navigationMenu.sharedNavButton().click()
+      sharedPage.sharedFolderItemName().contains(sharedFolderName)
+        .should("be.visible")
+        .dblclick()
+      sharedPage.fileItemRow().should("have.length", 1)
+
+      // ensure file name of moved file is correct
+      cy.get("@fileName").then(($fileName) => {
+        sharedPage.fileItemName().contains(`${$fileName}`).should("be.visible")
```
Comment (Tbaut) [inline packages/files-ui/cypress/tests/file-sharing-spec.ts]: A little more elegant IMHO. Feel free to ignore if you feel it's too complex
```suggestion
      cy.get<string>("@fileName").then((fileName) => {
        sharedPage.fileItemName().contains(fileName).should("be.visible")
```

**Review by Tbaut — APPROVED** [review]
Awesome! I left a couple nit suggestions

File: packages/files-ui/cypress/tests/file-sharing-spec.ts (hunk 6 above)
Comment (asnaith) [reply packages/files-ui/cypress/tests/file-sharing-spec.ts]: @Tbaut Thanks for catching these, I've updated all the comments :)

File: packages/files-ui/cypress/tests/file-sharing-spec.ts (hunk 9 above)
Comment (asnaith) [reply packages/files-ui/cypress/tests/file-sharing-spec.ts]: @Tbaut I like it. Thanks for the suggestion, I'm always thankful and appreciative of these tips to write cleaner code. 

I've gone ahead and used it as suggested and I also updated all the other places where we are doing similar things. It looks and reads better 👍

**Pull request merged**